lib/pages/timetable/view/events/event_view.dart

```diff
@@ -135,12 +135,12 @@ class _EventViewState extends State<EventView> {
                     child: Icon(FeatherIcons.mapPin),
                   ),
                   const SizedBox(width: 16),
-                  Text(widget.eventInstance.location,
+                  Text(widget.eventInstance?.location,
                       style: Theme.of(context).textTheme.subtitle1),
                 ],
               ),
             ),
-          if (widget.eventInstance.mainEvent != null)
+          if (mainEvent != null)
```

Comment:
This can never be true with the constructor asserts, so I think it's safe to remove the if here.